LLama/ChatSession.cs

```diff
@@ -494,3 +584,50 @@ in OutputTransform
         }
     }
 }
+
+/// <summary>
+/// The state of a chat session in-memory.
+/// </summary>
+public record SessionState
+{
+    /// <summary>
+    /// Saved executor state for the session in JSON format.
+    /// </summary>
+    public string ExecutorState { get; init; }
```

Comment:
I don't think we should use type `string` for it here. Though it's convenient to serialize the state and then save as file with a json format executor state, it doesn't seem to make sense in the case of keeping states in memory. It's better to serialize the executor state to string only when saving the `SessionState` to file.

Reply:
The reason I am using string here is that ExecutorState classes contain references to the internal state of the Executor (specifically the List variables I think).  So when Executor state is created and later executor is used again - ExecutorState get's updated unintentionally (by referencing internal state of the Executor).
Maybe it would make more sense to fix executor state creation and deep-copy those Lists/variables though?

Reply:
I prefer to making deep-copy here, because if we keep the data structure of `ExecutorState` here, it's convenient for users to modify the internal state. :) Currently there's no deep-copy method for `ExecutorState` so it may take you some extra time to make it.